src/components/ec/cuda/kernel/ec_cuda_reduce.cu

```diff
@@ -15,6 +15,7 @@ extern "C" {
 }
 #endif
 
+#include "ec_cuda_half_sm52.h"
```

Comment:
do we need to include this file in "ec_cuda_executor.cu" as well?

Reply:
Looks like "ec_cuda_executor.cu" is not doing any half operations, and the compiler is not complaining if I don't include this.

Reply:
actually it's used for PyTorch allreduce, from the backtrace 


2022-08-18T20:47:14.5411762Z [1660855634.540897] [0e25f152ba11:26395:0] ***ec_cuda_executor.cu:319***  UCC  ERROR cudaGetLastError()() failed: 209(no kernel image is available for execution on the device)

Reply:
I think the change in `cuda.m4` is sufficient for `ec_cuda_executor.cu`, and `ec_cuda_half_sm52.h` is needed for `https://ec_cuda_reduce.cu/`, because `no kernel image is available for execution on the device` only means the kernels in this file are not built for sm52, but it does not mean the kernels need fp16 arithmetic?

Reply:
There is no half here:
```cuda
__device__ void executor_reduce_task(ucc_eee_task_reduce_t *task)
{
    bool aligned = !(align_pow2((intptr_t)task->dst, 16) ||
                     align_pow2((intptr_t)task->srcs[0], 16) ||
                     align_pow2((intptr_t)task->srcs[1], 16));
    switch (task->dt) {
    case UCC_DT_FLOAT32:
        if (task->n_srcs == 2 && aligned) {
            executor_reduce_float_sum_aligned_2((float *)task->srcs[0],
                      (float *)task->srcs[1], (float *)task->dst, task->count);
        } else {
            executor_reduce_sum<float>((const float **)task->srcs, task->n_srcs,
                                       (float *)task->dst, task->count);
        }
        break;
    case UCC_DT_FLOAT64:
        executor_reduce_sum<double>((const double **)task->srcs, task->n_srcs,
                                    (double *)task->dst, task->count);
        break;
    case UCC_DT_INT32:
        executor_reduce_sum<int32_t>((const int32_t **)task->srcs, task->n_srcs,
                                     (int32_t *)task->dst, task->count);
        break;
    default:
        break;
    }
}
```

Reply:
you are right, it's sufficient now, will need to include this file when executor will support fp16